pieces/shared/menu-button/focus.tsx

```diff
@@ -0,0 +1,79 @@
+/**
+ * The focus method is only present on HTMLElement instances,
+ * so this convenience method simplifies dealing with different types.
+ */
+function focus(el: unknown) {
+  if (el instanceof HTMLElement) {
+    el.focus();
+  }
+}
+
+export function focusFirstItem(el: Element | null) {
+  focus(el?.firstElementChild);
+}
+
+export function focusLastItem(el: Element | null) {
+  focus(el?.lastElementChild);
+}
+
+export function focusNextItem(el: Element | null) {
+  const active = el?.querySelector('[tabindex="0"]');
+  if (active?.nextElementSibling) {
+    focus(active.nextElementSibling);
+  } else {
+    focusFirstItem(el);
+  }
+}
+
+export function focusPrevItem(el: Element | null) {
+  const active = el?.querySelector('[tabindex="0"]');
+  if (active?.previousElementSibling) {
+    focus(active.previousElementSibling);
+  } else {
+    focusLastItem(el);
+  }
+}
+
+/**
+ * When pressing a letter (A-Z or a-z) then:
```

Comment:
Is this _needed_ for accessibility? If not, I think we should drop this

Reply:
I cannot find exactly where in the W3 spec this is required, but we have it in our Accessibility Standards and also https://www.w3.org/WAI/ARIA/apg/example-index/menu-button/menu-button-actions.html.